pkg/apis/batch/validation/validation_test.go

```diff
@@ -1095,6 +1108,26 @@ func TestValidateJobUpdate(t *testing.T) {
 				Field: "spec.completionMode",
 			},
 		},
+		"Completions but not indexed jobs": {
+			old: batch.Job{
+				ObjectMeta: metav1.ObjectMeta{Name: "abc", Namespace: metav1.NamespaceDefault},
+				Spec: batch.JobSpec{
+					Selector:       validGeneratedSelector,
+					Template:       validPodTemplateSpecForGenerated,
+					CompletionMode: completionModePtr(batch.NonIndexedCompletion),
+					Completions:    pointer.Int32Ptr(2),
+				},
+			},
+			update: func(job *batch.Job) {
+				job.Spec.Completions = pointer.Int32Ptr(4)
+			},
+			err: &field.Error{
+				Type:  field.ErrorTypeInvalid,
+				Field: "spec.completions",
+			},
+			opts: JobValidationOptions{AllowElasticIndexedJobs: true},
```

Comment:
does this option matter?

Reply:
Yea it does.  Its not really an important test but this test case is verifying that we can't change completions if we are a NonIndexedJob.  I think scaling IndexJobs is gated by AllowElasticIndexedJobs.

Reply:
https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/batch/validation/validation.go#L570

If I don't have the AllowElasticIndexedJobs set we just assume completions are immutable.

Reply:
oh, then rename the testcase to: `immutable completions for indexed job when AllowElasticIndexedJobs is true`

Reply:
This test case is for non-indexed jobs but I take your point.  I'll push up a change.